src/plugins/tts/AbstractTTSEngine.js

```diff
@@ -147,36 +147,32 @@ export default class AbstractTTSEngine {
   }
 
   /** @private */
-  step() {
-    this._chunkIterator.next()
-      .then(chunk => {
-        if (chunk == PageChunkIterator.AT_END) {
-          this.stop();
-          this.opts.onDone();
-          return;
-        }
-
-        this.opts.onLoadingStart();
-        const sound = this.createSound(chunk);
-        sound.chunk = chunk;
-        sound.rate = this.playbackRate;
-        sound.voice = this.voice;
-        sound.load(() => this.opts.onLoadingComplete());
-
-        this.opts.onLoadingComplete();
-        return this.opts.beforeChunkPlay(chunk).then(() => sound);
-      })
-      .then(sound => {
-        if (!this.playing) return;
-
-        const playPromise = this.playSound(sound)
-          .then(() => this.opts.afterChunkPlay(sound.chunk));
-        if (this.paused) this.pause();
-        return playPromise;
-      })
-      .then(() => {
-        if (this.playing) return this.step();
-      });
+  async step() {
+    const chunk = await  this._chunkIterator.next();
+    if (chunk == PageChunkIterator.AT_END) {
+      this.stop();
+      this.opts.onDone();
+      return;
+    }
+    this.opts.onLoadingStart();
+    const sound = this.createSound(chunk);
+    sound.chunk = chunk;
+    sound.rate = this.playbackRate;
+    sound.voice = this.voice;
+    sound.load(() => this.opts.onLoadingComplete());
+
+    this.opts.onLoadingComplete();
+
+    const soundFn = await this.opts.beforeChunkPlay(chunk).then(() => sound);
+
+    if (!this.playing) return;
+
+    await this.playSound(soundFn);
+    this.opts.afterChunkPlay(soundFn.chunk);
+
+    if (this.paused) this.pause();
```

Comment:
Hmm, I can't remember exactly why this was the way it was, but I think we might need to keep it the same

```suggestion
    const playPromise = await this.playSound(sound).then(() => this.opts.afterChunkPlay(sound.chunk));
    if (this.paused) this.pause();
    await playPromise;
```